docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,220 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need a new api that will allow us to skip cluster/host validations, this will be for unsupported use.
+This API should work on a per cluster basis and will have it's own feature flag to allow us to enable it for specific users.
+An event should be emitted whenever the API is used during an installation.
+Clusters containing skipped validations should not contribute to statistics as these may be skewed.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- None identified at this stage
+
+## Proposal
+
+There are three main components as I see it
+
+### The REST API
+
+#### Endpoint: List existing host validations.
+
+GET api/v2/cluster/<id>/skipped-host-validations/
+Get a list of all skipped host validations.
+```
+{
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role"
+  ]
+}
+```
+#### Endpoint - List existing cluster validations.
+
+GET api/v2/cluster/<id>/skipped-cluster-validations/
+Get a list of all skipped cluster validations.
+```
+{
+    "network-prefix-valid",
+    "api-vips-defined"
+}
+```
+
+#### Endpoint - Add a skipped validation.
+
+PUT api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
+PUT api/v2/cluster/<id>/skipped-cluster-validations/api-vips-defined
```

Comment:
How do we plan the expose the list of available validations to users? I mean maybe we need also an API for fetching the 'host-validation-id' [enum](https://github.com/openshift/assisted-service/blob/58b7b4b0fd8e5a43c4b27ab28086c896fa636f27/swagger.yaml#L5836).

Reply:
Good question